### PR TITLE
feat: Records the start and end times of a track.

### PR DESCRIPTION
**Pull request by gpolitis**

### CODE REVIEW

File: src/features/quality-stats/QualityStatsCollector.js [hunk 1]
```diff
@@ -111,13 +114,17 @@ class QualityStatsCollector {
...
      * @param {string} ssrc - ssrc of the track from which we want to obtain collected data
+     * @param {number} timestamp - the UNIX timestamp of when this stats entry was observed on the client endpoint.
      * @returns {Object}`- Track collected data.
      */
-    _getTrackData(pcData, ssrc) {
-        if (!pcData[ssrc]) {
+    _getTrackData(pcData, ssrc, timestamp) {
+        if (pcData[ssrc]) {
+            // TODO what if endTime < startTime (client clock update)
+            pcData[ssrc].endTime = timestamp;
```
Comment (nils-ohlmeier) [inline src/features/quality-stats/QualityStatsCollector.js:123]: I would protect this with a check if `timestamp` is not undefined. That should prevent setting `endTime` accidentally. And as a plus allows to call `_getTrackData()` without a timestamp if needed.

I would compare the start and end time and rather set them to the same values, rather then creating negative track durations by allowing `endTime < startTime`.

File: src/features/quality-stats/StatsAggregator.js [hunk 2]
```diff
@@ -57,11 +57,13 @@ class StatsAggregator {
...
         };
 
-        const { packets, packetsLost, samplesReceived, concealedSamples } = trackStats;
+        const { packets, packetsLost, samplesReceived, concealedSamples, startTime, endTime } = trackStats;
 
         if (!packets.length) {
             return stats;
         }
+        stats.startTime = getSQLTimestamp(startTime);
+        stats.endTime = getSQLTimestamp(endTime);
```
Comment (nils-ohlmeier) [inline src/features/quality-stats/StatsAggregator.js:66]: I think this conversion is a level too high for my taste. I would prefer if this conversion happens in the Firehose connector. That would leave the option to write data into a non-SQL DB if needed.

**Review by nils-ohlmeier — CHANGES_REQUESTED** [review]
I think this PR is missing at least the changes needed in the Firehose SQL schema.

But based on my comments below I think it also missing change to the Firehose connector itself.

**Comment by andrei-gavrilescu** [comment]
LGTM!

**Pull request merged**